lte/gateway/c/core/oai/tasks/amf/amf_app_ue_context.cpp

```diff
@@ -429,55 +548,376 @@ void amf_free_ue_context(ue_m5gmm_context_s* ue_context_p) {
+    ue_context_p->cm_state = M5GCM_CONNECTED;
+
+    OAILOG_DEBUG_UE(
+        LOG_AMF_APP, ue_context_p->amf_context.imsi64,
+        "AMF_APP: UE Connection State changed to CONNECTED.enb_ue_s1ap_id "
+        "=" GNB_UE_NGAP_ID_FMT ", mme_ue_s1ap_id = " AMF_UE_NGAP_ID_FMT "\n",
+        ue_context_p->gnb_ue_ngap_id, ue_context_p->amf_ue_ngap_id);
+    // Set PPF flag to true whenever UE moves from M5GCM_IDLE to M5GCM_CONNECTED
+    // state
+    ue_context_p->ppf = true;
+
+    OAILOG_INFO_UE(LOG_AMF_APP, ue_context_p->amf_context.imsi64,
+                   "UE STATE - CONNECTED.\n");
+  } else if (ue_context_p->cm_state == M5GCM_IDLE &&
+             new_cm_state == M5GCM_IDLE) {
+    OAILOG_INFO_UE(
+        LOG_AMF_APP, ue_context_p->amf_context.imsi64,
+        "Old UE CM State (IDLE) is same as the new UE CM state (IDLE)\n");
+    magma::map_rc_t m_rc = magma::MAP_OK;
+    m_rc = amf_ue_context_p->gnb_ue_ngap_id_ue_context_htbl.remove(
+        ue_context_p->gnb_ngap_id_key);
+
+    if (m_rc != magma::MAP_OK) {
+      OAILOG_WARNING_UE(LOG_AMF_APP, ue_context_p->amf_context.imsi64,
+                        "UE context gnb_ue_ngap_ue_id_key %ld "
+                        "amf_ue_ngap_id " AMF_UE_NGAP_ID_FMT
+                        ", GNB_UE_NGAP_ID_KEY could not be found",
+                        ue_context_p->gnb_ngap_id_key,
+                        ue_context_p->amf_ue_ngap_id);
+    }
+
+    ue_context_p->gnb_ngap_id_key = INVALID_GNB_UE_NGAP_ID_KEY;
+  } else {
+    OAILOG_INFO_UE(LOG_AMF_APP, ue_context_p->amf_context.imsi64,
+                   "Old UE CM State (CONNECTED) is same as the new UE CM state "
+                   "(CONNECTED)\n");
+  }
+  OAILOG_FUNC_OUT(LOG_AMF_APP);
+}
+
+// Context release timer expiry
+static int amf_ue_context_release_complete_timer_handler(zloop_t* loop,
+                                                         int timer_id,
+                                                         void* output) {
+  amf_context_t* amf_ctx = NULL;
+  amf_ue_ngap_id_t ue_id = 0;
+
+  if (!amf_pop_timer_arg(timer_id, &ue_id)) {
+    OAILOG_WARNING(
+        LOG_AMF_APP,
+        "Context Release Timer: Invalid Timer Id expiration, Timer Id: %u\n",
+        timer_id);
+    OAILOG_FUNC_RETURN(LOG_NAS_AMF, RETURNok);
   }
 
-  if (ue_context_p->amf_ue_ngap_id != INVALID_AMF_UE_NGAP_ID) {
-    m_rc = amf_state_ue_id_ht->remove(ue_context_p->amf_ue_ngap_id);
-    if (m_rc != magma::MAP_OK)
-      OAILOG_TRACE(LOG_AMF_APP, "Error Could not remove this ue context \n");
-    ue_context_p->amf_ue_ngap_id = INVALID_AMF_UE_NGAP_ID;
+  ue_m5gmm_context_s* ue_amf_context =
+      amf_ue_context_exists_amf_ue_ngap_id(ue_id);
+
+  if (ue_amf_context == NULL) {
+    OAILOG_ERROR(LOG_NAS_AMF,
+                 "Ue Context Release Timer: ue_amf_context is NULL for "
+                 "ue id: " AMF_UE_NGAP_ID_FMT "\n",
+                 ue_id);
+    OAILOG_FUNC_RETURN(LOG_NAS_AMF, RETURNok);
```

Comment:
Module name should be LOG_AMF_APP.
General comment is to use appropriate module name

Reply:
Comment addressed.